### PR TITLE
KAFKA-15816: Fix leaked sockets in streams tests

### PR DESCRIPTION
**Pull request by gharris1727**
These tests leak sockets via Kafka clients, either directly or via KafkaStreams instances. The tests should close these resources when appropriate.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/integration/IQv2IntegrationTest.java [hunk 1]
```diff
@@ -418,6 +418,8 @@ public String metricsScope() {
             })
         );
 
+        // do not use the harness streams
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/IQv2IntegrationTest.java]: Not sure what this comment means?

File: streams/src/test/java/org/apache/kafka/streams/integration/MetricsReporterIntegrationTest.java [hunk 2]
```diff
@@ -125,6 +126,8 @@ public void shouldBeAbleToProvideInitialMetricValueToMetricsReporter() {
             final Object initialMetricValue = METRIC_NAME_TO_INITIAL_VALUE.get(metricName.name());
             assertThat(initialMetricValue, notNullValue());
         });
+
+        kafkaStreams.close(Duration.ofSeconds(30));
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/MetricsReporterIntegrationTest.java]: Should we use `try-with-resources` ? Do we need the timeout parameter (if yes, might still be better to use a try-catch with `finally` block?

**Review by mjsax — COMMENTED** [review]
Thanks for the PR! Very nice to get these leaks fixed.

File: streams/src/test/java/org/apache/kafka/streams/integration/IQv2IntegrationTest.java (hunk 1 above)
Comment (gharris1727) [reply streams/src/test/java/org/apache/kafka/streams/integration/IQv2IntegrationTest.java]: I changed this, let me know if the new comment makes more sense or if you'd like it removed.

I was indicating that the KafkaStreams that the test harness set up for the test won't be used, and a different KafkaStreams was being substituted.

**Review by mjsax — APPROVED** [review]
Thanks. LGTM.

**Comment by gharris1727** [comment]
The streams test failures appear unrelated, as they are not subclasses of the changed tests, and all streams tests pass locally.

**Pull request merged**